OsmAnd/src/net/osmand/plus/routing/RoutingHelper.java

```diff
@@ -502,6 +531,20 @@ private boolean updateCurrentRouteStatus(Location currentLocation, float posTole
 				route.updateCurrentRoute(newCurrentRoute + 1);
 				currentRoute = newCurrentRoute + 1;
 				app.getNotificationHelper().refreshNotification(NotificationType.NAVIGATION);
+				if (!updateListeners.isEmpty()) {
+					ArrayList<WeakReference<IRoutingDataUpdateListener>> tmp = new ArrayList<>(updateListeners);
```

Comment:
Here no need to modify list! Could be very slow!

Reply:
+